main/inc/lib/userportal.lib.php

```diff
@@ -1165,11 +1165,118 @@ public function returnCoursesAndSessions($user_id)
                 $this->load_directories_preview
             );
 
+
+            //Course option (show student progress)
+            //This code will add a new variables (Progress, Score, Certificate)
+            if(api_get_configuration_value('my_course_progress') === true && api_is_student() === true) {
+                
+                foreach($specialCourses as $key => $specialCourseInfo) {
+                    $progress = Tracking::get_avg_student_progress(
+                        $user_id,
+                        $specialCourseInfo['course_code']
+                    );
+                    $percentage_score = Tracking::get_avg_student_score(
+                        $user_id,
+                        $specialCourseInfo['course_code'],
+                        array()
+                    );
+                    $specialCourses[$key]['student_info']['progress'] = ($progress === false)? null : $progress;
+                    $specialCourses[$key]['student_info']['score'] = $percentage_score;
+    
+                    $category = Category::load(
+                        null,
+                        null,
+                        $specialCourseInfo['course_code'],
+                        null,
+                        null,
+                        null
+                    );
+                    $specialCourses[$key]['student_info']['certificate'] = null;
+                    if (isset($category[0])) {
+                        if($category[0]->is_certificate_available($user_id)) {
+                            $specialCourses[$key]['student_info']['certificate'] = Display::label(get_lang('Yes'), 'success');
+                        } else {
+                            $specialCourses[$key]['student_info']['certificate'] = Display::label(get_lang('No'));
+                        }
+                    }
+                }
+                
+                if(isset($courses['in_category']) && isset($courses['not_category'])){
+                    foreach($courses['in_category'] as $key1 => $value) {
+                        if(isset($courses['in_category'][$key1]['courses'])){
+                            foreach($courses['in_category'][$key1]['courses'] as $key2 => $courseInCatInfo) {
```

Comment:
Same spacing observation (plus same observation for the end of the line: missing space before "{") for these 4 lines